src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs

```diff
@@ -25,14 +26,56 @@ public static void EnsureDotnet()
             }
         }
 
+        public static async Task<int> GetDotnetMajorVersion()
+        {
+            EnsureDotnet();
+            Version dotnetVersion;
+            var exe = new Executable("dotnet", "--version");
+            var stdout = new StringBuilder();
+            var exitCode = await exe.RunAsync(o => stdout.AppendLine(o), e => ColoredConsole.Error.WriteLine(ErrorColor(e)));
+            // We split based on the the '-' character and take the first element to ensure that version parsing still works with preview/RC verisons of dotnet
+            var dotnetVersionString = stdout.ToString().Split("-")[0];
+            if (exitCode != 0)
+            {
+                throw new CliException("Error running dotnet --version");
+            }
+            else if (!Version.TryParse(dotnetVersionString, out dotnetVersion))
+            {
+                throw new CliException("dotnet version unable to be parsed!");
+            }
+            return dotnetVersion.Major;
+        }
+
+        public static async Task<string> GetTargetFramework()
+        {
+            var framework = "net";
+            var dotnetMajorVersion = await GetDotnetMajorVersion();
+            if (dotnetMajorVersion == 7)
+            {
+                framework += "7.0";
+            }
+            else if (dotnetMajorVersion == 6)
+            {
+                framework += "6.0";
+            }
+            else
+            {
+                framework += "48";
```

Comment:
I think we should default to 6.0 to avoid any potential breaking changes, by that I mean adding an explicit check for the major version being '48' and the default case being 6 (a switch statement would work well for this)